### PR TITLE
*: remove engine_rocks raw_util

### PR DESCRIPTION
**Pull request by BusyJay**
### What is changed and how it works?

Issue Number: Ref #13058 

What's Changed:

```commit-message
raw_util is duplicated with util. By using util, this PR also makes `DB`
type only in use in engine_rocks, which is helpful for adapting tirocks
later.

In addition, this PR fixes an unsound transforms between `Arc<DB>` and
`RocksEngine`.
```

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/backup/src/endpoint.rs [hunk 1]
```diff
@@ -505,7 +505,7 @@ impl BackupRange {
     async fn backup_raw_kv_to_file<E: Engine>(
         &self,
         engine: E,
-        db: Arc<DB>,
+        db: RocksEngine,
```
Comment (Connor1996) [inline components/backup/src/endpoint.rs:508]: why it's generic on Engine, but still need to pass a rocksdb?

Comment (BusyJay) [reply components/backup/src/endpoint.rs:508]: /cc @3pointer

**Review by Connor1996 — APPROVED** [review]
rest LGTM

File: components/engine_rocks/src/cf_options.rs [hunk 2]
```diff
@@ -27,28 +29,40 @@ impl CFOptionsExt for RocksEngine {
     }
 }
 
-#[derive(Clone)]
-pub struct RocksColumnFamilyOptions(RawCFOptions);
+#[derive(Default, Clone)]
+pub struct RocksCfOptions(RawCfOptions);
```
Comment (tabokie) [inline components/engine_rocks/src/cf_options.rs:33]: Bit weird to have both `RocksCfOptions` and `CFOptionsExt`/`RocksDBOptions`.

**Review by tabokie — APPROVED** [review]

File: components/engine_rocks/src/cf_options.rs (hunk 2 above)
Comment (BusyJay) [reply components/engine_rocks/src/cf_options.rs:33]: I will rename all those struct in a separate PR.

File: components/backup/src/endpoint.rs (hunk 1 above)
Comment (BusyJay) [reply components/backup/src/endpoint.rs:508]: I merge it now, and solve issues further if there is any.

Comment (3pointer) [reply components/backup/src/endpoint.rs:508]: We need build `RocksSstWriterBuilder` from `RocksEngine` during backup. but it seems we cannot get `RocksEngine` from generic Engine 🤔

Comment (Connor1996) [reply components/backup/src/endpoint.rs:508]: maybe you don't need to call `set_db` for `RocksSstWriterBuilder`

Comment (3pointer) [reply components/backup/src/endpoint.rs:508]: if we don't call  `set_db`,  it seems we cannot get `cf_options` of current rocksdb. this makes backup sst different with normal sst files.

**Pull request merged**